src/Appwrite/Auth/OAuth2/Dailymotion.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+namespace Appwrite\Auth\OAuth2;
+
+use Appwrite\Auth\OAuth2;
+
+// Reference Material
+// https://developers.dailymotion.com/api/#authentication
+
+class Dailymotion extends OAuth2
+{
+    /**
+     * @var string
+     */
+    private string $endpoint = 'https://api.dailymotion.com';
+
+    /**
+     * @var string
+     */
+    private string $authEndpoint = 'https://www.dailymotion.com/oauth/authorize';
+
+    /**
+     * @var array
+     */
+    protected array $scopes = [
+        'userinfo',
+        'email'
+    ];
+
+    /**
+     * @var array
+     */
+    protected array $fields = [
```

Comment:
by default, the `getUser()` function returns only `id` and `screen_name`. We need to add more fields for this to get more info

Reply:
However, I am unsure that do we need all those fields or some of them are redundant?

Reply:
We only need ID, email, and first and last name.

Username for sure can be removed. I don't think we fallback to it in any provider.

Verified should be saying if the user has confirmed their Dailymotion account email.. This is not used right now but will be in the future. So let's keep it.

Status can probably be removed. We don't really care if they are banned or whatever, as long as they can sign in, we are satisfied.